### PR TITLE
Implement codefix for MSTEST0006 (AvoidExpectedException)

### LINKED ISSUE #3399: MSTEST0006: Add code fix
AB#2200928

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #3399

### CODE REVIEW

**Comment by Youssef1313** [comment]
@Evangelink Current implementation only wraps last statement. Let me know if we want to implement both fixes or if we want to go with this one first until we hear user feedback

**Comment by Evangelink** [comment]
I think that will be the most used code fix so let's ship this one only for now and wait for feedback. Please once this is merged, update the documentation https://learn.microsoft.com/dotnet/core/testing/mstest-analyzers/mstest0006 to mention we have code fix and the version it's introduced into

File: src/Analyzers/MSTest.Analyzers.CodeFixes/AvoidExpectedExceptionAttributeFixer.cs
Comment (Evangelink) [inline src/Analyzers/MSTest.Analyzers.CodeFixes/AvoidExpectedExceptionAttributeFixer.cs:117]: Isn't it easier/faster to just look if method declaration as async?

File: test/UnitTests/MSTest.Analyzers.UnitTests/AvoidExpectedExceptionAttributeAnalyzerTests.cs
Comment (Evangelink) [inline test/UnitTests/MSTest.Analyzers.UnitTests/AvoidExpectedExceptionAttributeAnalyzerTests.cs]: Let's link this issue https://github.com/microsoft/testfx/issues/4034 and say the expected diagnostics should be gone when it's fixed.

**Review by Evangelink — COMMENTED** [review]
Could you please add a test, sync and async (it's fine to do both in the same test) with multiple statements to clearly document we only wrap last statement?

File: src/Analyzers/MSTest.Analyzers.CodeFixes/AvoidExpectedExceptionAttributeFixer.cs
Comment (Evangelink) [inline src/Analyzers/MSTest.Analyzers.CodeFixes/AvoidExpectedExceptionAttributeFixer.cs]: It should be updated to be based on #4039

File: src/Analyzers/MSTest.Analyzers.CodeFixes/AvoidExpectedExceptionAttributeFixer.cs
Comment (Youssef1313) [reply src/Analyzers/MSTest.Analyzers.CodeFixes/AvoidExpectedExceptionAttributeFixer.cs:117]: @Evangelink The test method could be async, but the specific line being wrapped in ThrowsException may be synchronous. I'll add a test to cover this case and make sure it doesn't regress.

Something along the lines of

```csharp
[TestMethod]
[ExpectedException(typeof(Exception))]
public async Task MyTestMethod()
{
    await Task.Delay(100);
    MyMethod();
}

private static void MyMethod() => throw new Exception();
```

It should just be `Assert.ThrowsException<Exception>(() => MyMethod);` and not ThrowsExceptionAsync.

It goes the other way around where we need to use Assert.ThrowsExceptionAsync if the last statement in the test had the `await` deeper, e.g, `MyMethod(await MyOtherMethod())`

File: test/UnitTests/MSTest.Analyzers.UnitTests/AvoidExpectedExceptionAttributeAnalyzerTests.cs
Comment (Youssef1313) [reply test/UnitTests/MSTest.Analyzers.UnitTests/AvoidExpectedExceptionAttributeAnalyzerTests.cs]: It's not related. #4034 is about work to be done on the analyzer. The test here uses ExpectedExceptionAttribute which is supported. However, what is not supported (intentionally) by the codefix here is when the attribute has `AllowDerivedTypes = true` because using `Assert.ThrowsException<ExceptionType>(...)` isn't going to preserve the semantics. The user needs to specify the exact exception type, and not a base one.

File: src/Analyzers/MSTest.Analyzers.CodeFixes/AvoidExpectedExceptionAttributeFixer.cs
Comment (Evangelink) [reply src/Analyzers/MSTest.Analyzers.CodeFixes/AvoidExpectedExceptionAttributeFixer.cs:117]: Definitely! Sorry I overlooked that case.

**Review by Evangelink — APPROVED** [review]

**Pull request merged**
